### ISSUE TITLE
segfault in cmrc/test_ucp_rma.nonblocking_get_nbi_flush_worker

### ISSUE THREAD

**Issue by yosefe** [openucx/ucx#742]
http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1202/label=hpc-test-node/console
```
19:59:01 [       OK ] cmrc/test_ucp_rma.nonblocking_get_nbi_flush_worker/0 (164 ms)
19:59:01 [ RUN      ] cmrc/test_ucp_rma.nonblocking_stream_put_nbi_flush_worker/0
19:59:01 [hpc-test-node:3287 :0] Caught signal 11 (Segmentation fault: Sent by the kernel at address (nil))
19:59:01 ./contrib/test_jenkins.sh: line 196:  3259 User defined signal 1   $AFFINITY $TIMEOUT make -C test/gtest test UCS_HANDLE_ERRORS=bt19:59:01 [       OK ] cmrc/test_ucp_rma.nonblocking_get_nbi_flush_worker/0 (164 ms)
19:59:01 [ RUN      ] cmrc/test_ucp_rma.nonblocking_stream_put_nbi_flush_worker/0
19:59:01 [hpc-test-node:3287 :0] Caught signal 11 (Segmentation fault: Sent by the kernel at address (nil))
19:59:01 ./contrib/test_jenkins.sh: line 196:  3259 User defined signal 1   $AFFINITY $TIMEOUT make -C test/gtest test UCS_HANDLE_ERRORS=bt
```

**Comment by yosefe** [comment]
Another similar failure from ORNL:
```
[ RUN      ] udrc/test_ucp_tag_match.send2_nb_recv_medium_wildcard/0
[     INFO ] Testing unexpected mode, size 3000000
[     INFO ] Testing expected mode, size 3000000
[pod9:20063:3] rc_verbs_iface.c:98   Fatal: Send completion with error: transport retry counter exceeded
[pod9:20063:3] Process frozen...
```

**Comment by yosefe** [comment]
fixed by #744
